flask_restless/helpers.py

```diff
@@ -234,10 +235,20 @@ def query_by_primary_key(session, model, pk_value, primary_key=None):
     If `primary_key` is specified, the column specified by that string is used
     as the primary key column. Otherwise, the column named ``id`` is used.
 
+    Convert the pk_value to int if primary_key type is Integer.
+
     Presumably, the returned query should have at most one element.
 
     """
-    pk_name = primary_key
+    pk_name = primary_key if primary_key else 'id'
+    pk_type = get_field_type(model, pk_name)
+
+    try:
+        if isinstance(pk_type, Integer):
```

Comment:
nit: could you please move try..except inside `if`? As it only applies to `pk_value = int(pk_value)` operation

Reply:
Sure :)